warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java

```diff
@@ -215,6 +217,65 @@ public ScannerChunk apply(@Nullable ScannerChunk input) {
                         }
                     }
                     
+                    int termCount = TermCountingVisitor.countTerms(script);
+                    if (termCount > config.getMaxTermThreshold()) {
+                        if (canIvarate(config, setting)) {
+                            // copy the config and set minimum thresholds for collapsing Ors
+                            ShardQueryConfiguration reduceConfig = new ShardQueryConfiguration(config);
+                            reduceConfig.setMaxOrExpansionThreshold(2);
+                            reduceConfig.setMaxOrRangeThreshold(2);
+                            
+                            try {
+                                Query settings = config.getQuery();
+                                
+                                if (config.canHandleExceededValueThreshold()) {
+                                    URI hdfsQueryCacheUri = getFstHdfsQueryCacheUri(config, settings);
+                                    SortedMap<Integer,List<String>> pushdownCapacity;
+                                    FileSystem fs = null;
+                                    String fstHdfsUri = null;
+                                    if (hdfsQueryCacheUri != null) {
+                                        fs = VisitorFunction.fileSystemCache.getFileSystem(hdfsQueryCacheUri);
+                                        fstHdfsUri = hdfsQueryCacheUri.toString();
+                                    }
+                                    
+                                    pushdownCapacity = PushdownLargeFieldedListsVisitor.getPushdownCapacity(reduceConfig, script, fs, fstHdfsUri);
```

Comment:
Instead of calculating this on the second pass of the PushdownLargeFieldedListsVisitor, couldn't we just generate this on the first pass for those terms which didn't get combined into a list ivarator?  Might save some time.